src/main/java/decodes/tsdb/algo/AverageAlgorithm.java

```diff
@@ -1,84 +1,61 @@
 package decodes.tsdb.algo;
 
-import java.util.Date;
-
-import ilex.var.NamedVariableList;
 import ilex.var.NamedVariable;
-import decodes.tsdb.DbAlgorithmExecutive;
 import decodes.tsdb.DbCompException;
-import decodes.tsdb.DbIoException;
-import decodes.tsdb.VarFlags;
-import decodes.util.PropertySpec;
+import org.opendcs.annotations.PropertySpec;
+import org.opendcs.annotations.algorithm.Algorithm;
+import org.opendcs.annotations.algorithm.Input;
+import org.opendcs.annotations.algorithm.Output;
 
-//AW:JAVADOC
-/**
-AverageAlgorithm averages single 'input' parameter to a single 'average' 
-parameter. The averaging period is determined by the interval of the output
-parameter.
+@Algorithm(
+		description = "AverageAlgorithm averages single 'input' parameter to a single 'average' \n" +
+				"parameter. The averaging period is determined by the interval of the output\n" +
+				"parameter."
+)
 
- */
-//AW:JAVADOC_END
 public class AverageAlgorithm extends decodes.tsdb.algo.AW_AlgorithmBase
 {
-//AW:INPUTS
-	double input;	//AW:TYPECODE=i
-	String _inputNames[] = { "input" };
-//AW:INPUTS_END
+	private static final String AVERAGESTRING = "average";
+	@Input
+	double input;
 
-//AW:LOCALVARS
 	double tally;
 	int count;
-	PropertySpec specs[] = 
-	{
-		new PropertySpec("negativeReplacement", PropertySpec.NUMBER, 
-			"(no default) If set, and output would be negative, then replace with the number supplied.")
-	};
 
-//AW:LOCALVARS_END
+	@Output
+	NamedVariable average = new NamedVariable(AVERAGESTRING, 0);
 
-//AW:OUTPUTS
-	NamedVariable average = new NamedVariable("average", 0);
-	String _outputNames[] = { "average" };
-//AW:OUTPUTS_END
-
-//AW:PROPERTIES
+	@org.opendcs.annotations.PropertySpec(value = "1")
 	public long minSamplesNeeded = 1;
+	@PropertySpec(value = "-1.0 / 0.0")
```

Comment:
I don't believe our system will do this math and thus this may break usages.

Don't worry about it for now though, I'll look into that a bit later.

Reply:
I just looked, the properties system will handled the text `Double.NEGATIVE_INIFINITY` correct, use that exact text as the default value.